### PR TITLE
added Hint Option Menu item

### PR DESCRIPTION
**Pull request by mmaulwurff**
Hint is a static text that is visible only if the items above are selected.
The number of items above is configurable, and by default is 1.

Hint Demo: https://www.mediafire.com/file/4p1s087fpsv28fb/hint-demo.pk3/file

### CODE REVIEW

File: wadsrc/static/zscript/ui/menu/optionmenuitems.zs [hunk 1]
```diff
@@ -1253,3 +1253,47 @@ class VideoModeMenu : OptionMenu
...
+	private bool isAboveSelected(OptionMenuDescriptor desc)
+	{
+		int selfIndex = desc.mItems.find(self);
+		int aboveEnd  = selfIndex - 1;
+
+		while (aboveEnd >= 0 && !desc.mItems[aboveEnd].Selectable()) { --aboveEnd; }
+
+		int  aboveBegin      = aboveEnd - mAboveNumber + 1;
+		int  selected        = desc.mSelectedItem;
+		console.printf("selected: %d", selected);
```
Comment (alexey-lysiuk) [inline wadsrc/static/zscript/ui/menu/optionmenuitems.zs]: Please remove this line.

Comment (mmaulwurff) [reply wadsrc/static/zscript/ui/menu/optionmenuitems.zs]: Done.

**Comment by mmaulwurff** [comment]
Some mods start to use hints in their menus (for example, Nash's Gore Mod: https://forum.zdoom.org/viewtopic.php?f=46&t=62641#p1080654). This means that this kind of feature is useful for modders.

There was a request for this in Brainstorming a GZDoom menu revamp (https://forum.zdoom.org/search.php?keywords=description&t=60041&sf=msgonly), so this may be useful in engine, too.

**Comment by mmaulwurff** [comment]
Graf Zahl

> As implemented the feature is not acceptable. This isn't how hints should be handled. It cannot just be done by adding new menu entries, you'll really have to make some alteration to the menu itself and look out for how it affects customized subclasses (e.g. the flag display for the gameplay and compatibility menus.)

Noted.

**Pull request closed without merging**